source/frontend/src/components/common/HeaderField/StatusField.tsx

```diff
@@ -0,0 +1,138 @@
+}
+
+const statusDictionary: Dictionary<StatusStyle> = {
+  active: { icon: MdCircle, colorVariant: 'green' },
+  archived: { icon: MdArchive, colorVariant: 'grey' },
+  completed: { icon: FaCheck, colorVariant: 'blue' },
+  hold: { icon: FaPauseCircle, colorVariant: 'yellow' },
+  cancelled: { icon: MdCancel, colorVariant: 'red' },
+  draft: { icon: MdEdit, colorVariant: 'yellow' },
+  terminated: { icon: FaStopCircle, colorVariant: 'red' },
+  duplicate: { icon: BiDuplicate, colorVariant: 'blue' },
+};
+
+const StatusField: React.FunctionComponent<React.PropsWithChildren<IStatusFieldProps>> = ({
+  statusCodeType,
+}) => {
+  const translateStatusCode = (statusCodeType: ApiGen_Base_CodeType<string>) => {
+    switch (statusCodeType.id.toLowerCase()) {
+      case 'active':
+      case 'ac':
+        return statusDictionary['active'];
+      case 'archived':
+      case 'cncn':
+      case 'archiv':
+        return statusDictionary['archived'];
+      case 'cancelled':
+      case 'ca':
+      case 'cancel':
+      case 'discard':
+        return statusDictionary['cancelled'];
+      case 'complete':
+      case 'closed':
+      case 'co':
+      case 'complt':
+        return statusDictionary['completed'];
+      case 'draft':
+      case 'pl':
+        return statusDictionary['draft'];
+      case 'duplicate':
+        return statusDictionary['duplicate'];
+      case 'hold':
+      case 'inactive':
+      case 'ho':
+        return statusDictionary['hold'];
+      case 'terminated':
+        return statusDictionary['terminated'];
+      default:
+        break;
+    }
+    return statusDictionary['cancelled'];
+  };
+
+  const statusFound = translateStatusCode(statusCodeType);
+  return (
+    <StyledBottomRow className="no-gutters justify-content-end align-items-end">
+      <Col />
+      <Col xs="auto" className="align-self-end">
+        <RetiredWarning $variant={statusFound.colorVariant}>
+          <statusFound.icon size={16} />
+          {statusCodeType?.description.toUpperCase()}
+        </RetiredWarning>
+      </Col>
+    </StyledBottomRow>
+  );
+};
+
+export default StatusField;
+
+const RetiredWarning = styled(InlineFlexDiv)<{ $variant: string }>`
+  //font-weight: bold; TODO: Bold should be enough, but atm the bcsans-bold is different
+  font-family: 'BCSans-Bold';
+  font-size: 1.4rem;
+
+  border-radius: 0.4rem;
+
+  letter-spacing: 0.1rem;
+  align-items: center;
+
+  padding: 0.2rem 0.5rem;
+
+  gap: 0.5rem;
+
+  ${props => {
+    if (props.$variant === 'green') {
+      return `
+  color: ${props.theme.css.completedColor};
+  background-color: ${props.theme.css.selectedColor};
+    `;
+    } else if (props.$variant === 'grey') {
+      return `
+  color: ${props.theme.css.disabledColor};
+  background-color: ${props.theme.css.disabledFieldBackgroundColor};
+    `;
+    } else if (props.$variant === 'blue') {
+      return `
+  color: ${props.theme.css.slideOutBlue};
+  background-color: ${props.theme.css.filterBoxColor};
```

Comment:
I'm confused about the naming choices here. It seems like you could have named these colors whatever you wanted, but ended up going with some of the previously used colors that have since been removed.

For example, why call the blue slideOutBlue? I recognize that as a color that we used to have, but if this is the only incidence of this color, why not name this something closer to the usage. Ie. 'completedStatusColor' or similar.

Reply:
tbh I had a hard time with the colors and the documentation on figma so I decided to name it exactly like the figma ones . I will chat with ana when she is back to get those names aligned.